### ISSUE TITLE
buildenv is slow

### ISSUE THREAD

**Issue by mkenigs** [flox/flox#1078]
For an environment with a lot of packages, building takes me 30 seconds, even if we've already built the environment once before.

```
> flox list
alejandra: alejandra (3.0.0)
bashInteractive: bashInteractive (5.2)
difftastic: difftastic (0.53.0)
direnv: direnv (2.32.3)
git: git (2.42.0)
gnupg: gnupg (2.4.1)
iterm2: iterm2 (3.4.22)
jq: jq (1.7.1)
neovim: neovim (0.9.4)
openssh: openssh (9.6p1)
ripgrep: ripgrep (13.0.0)
shellcheck: shellcheck (0.9.0)
tree: tree (2.1.1)
> rm .flox/run/aarch64-darwin.default && time flox activate -- true
flox activate -- true  0.02s user 0.01s system 0% cpu 31.662 total
> rm .flox/run/aarch64-darwin.default && time flox activate -- true
flox activate -- true  0.02s user 0.01s system 0% cpu 32.826 total
```

Part of the problem is that we're building twice, which we'll address in https://github.com/flox/flox/issues/1062

But even if we only build once that's still ~15 seconds.

AC: buildenv for a lot of packages that have previously been installed only takes a few seconds.

**Comment by zmitchell** [comment]
This is addressed by #1094

**Comment by mkenigs** [comment]
🔥 
```
time flox activate -- true
flox activate -- true  0.02s user 0.02s system 8% cpu 0.476 total
```
